include/zephyr/kernel/thread.h

```diff
@@ -237,7 +237,7 @@ typedef struct k_thread_runtime_stats {
 	uint64_t idle_cycles;
 #endif /* CONFIG_SCHED_THREAD_USAGE_ALL */
 
-#if defined(__cplusplus) && !defined(CONFIG_SCHED_THREAD_USAGE) &&                                 \
+#if (defined(__cplusplus) || CONFIG_RUST) && !defined(CONFIG_SCHED_THREAD_USAGE) &&	\
```

Comment:
Same comment from elsewhere: does Rust not allow this either?  Should probably update the comment if so.

Reply:
I'm not sure if this needs to be fixed in bindgen, or exactly what the best place to fix it is. Basically, Rust pretty readily supports zero-size items (they are used quite frequently), but considers them not FFI-safe since they aren't technically allowed by C. Maybe a longer term solution would be to inform rust that the C compiler involved supports zero length structs and this shouldn't generate this warning. It might also be possible to turn the warning off, but I would like to generate a warning with structs that are truly non FFI-safe, such as fat pointers and other features Rust has.

Reply:
C is fine, it's a C++ issue. But Rust probably tries to be safe to use with C++ 🫠

Reply:
Actually, C is not fine with it. It is an extension that is supported, as far as I know, by both gcc and clang, but is certainly not universal. For example, apparently the MSVC compiler does not support them. That isn't likely something we need in Zephyr, but the Rust compiler intends to support interoperation with this tool

I can look at what might be possible as far as suppressing the specific warning in the Rust bindgen generated code.

Reply:
See above, I've reverted this change, and will deal with the warning on the Rust code.